src/perl.c

```diff
@@ -1923,6 +1923,11 @@ static int perl_read (void)
 		aTHX = t->interp;
 	}
 
+	/* Assert that we're not running as the base thread. Otherwise, we might
+	 * run into concurrency issues with c_ithread_create(). See
+	 * https://github.com/collectd/collectd/issues/9 for details. */
+	assert (aTHX != perl_threads->head->interp);
```

Comment:
This line breaks when collectd run in single thread (-f commandline option or smth like).